clamonacc/clamav-clamonacc.service.in

```diff
@@ -10,8 +10,30 @@ After=clamav-daemon.service syslog.target network.target
 [Service]
 Type=simple
 User=root
-ExecStartPre=/bin/bash -c "while [ ! -S /run/clamav/clamd.ctl ]; do sleep 1; done"
-ExecStart=@prefix@/sbin/clamonacc -F --log=/var/log/clamav/clamonacc.log --move=/root/quarantine
+ExecStartPre=/usr/bin/install --owner=root --group=root --directory /var/local/quarantine
```

Comment:
I find that this fails if `/var/local/quarantine` does not exist.  

I created it and did `sudo chown clamav /var/local/quarantine` and then tried again. This time it failed with `Failed to execute /usr/bin/install: Permission denied", which seems similar to the failure I'm seeing with trying to start cthe clamd service. 

I'm at a loss as the ExecPaths in both service files look correct to me.

Reply:
Thanks, let me investigate.

Reply:
Can you give me more information about your setup so I can attempt to replicate the issue?

Reply:
I'm testing on an ubuntu 22.04 VM where I've installed `systemd` and `libsystemd-dev` before [building clamav with these instructions](https://docs.clamav.net/manual/Installing/Installing-from-source-Unix.html#ubuntu--debian) and installing to the default prefix.  Like:
`mkdir build && cd build && cmake .. && make -j12 && sudo make install`

I had to set up:
- `/usr/local/etc/freshclam.conf` and `/usr/local/etc/clamd.conf` with the basics
- `sudo groupadd clamav && sudo useradd -g clamav -s /bin/false -c "Clam Antivirus" clamav`
- `sudo mkdir /usr/local/share/clamav && sudo chown clamav /usr/local/share/clamav`
- `sudo mkdir /var/log/clamav && sudo chown clamav /var/log/clamav`

I think that's it. Then after reloading with `sudo systemctl daemon-reload`, I tried starting the various services.

Reply:
> Can you give me more information about your setup so I can attempt to replicate the issue?

You did a lot of work on this. I don't want to close/cancel this PR unless you've lost interest in solving the remaining issues.

Reply:
@micahsnyder Sorry, this completely slipped my mind. I'll work on this over the weekend. I already have a VM that I can test this in.

Reply:
Setting DynamicUser=yes results in "Can't save PID to file /run/clamav/clamd.pid: Permission denied"

Reply:
To begin with, I tested the default installation on Debian 12. I found one issue where if `/usr/lib/x86_64-linux-gnu` had to be in the `ExecPaths` or clamav-daemon won't start. I also found `/dev/null` in the list of open files by the daemon - preferably for writing. I don't know if `/dev/null` and other char-files need to be in the read-only exception in the unit file.